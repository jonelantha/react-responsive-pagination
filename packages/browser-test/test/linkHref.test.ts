import { TestHarnessPage } from './test-harness-page';

const testHarness = new TestHarnessPage(page, { throwOnError: true });

beforeAll(async () => {
  await testHarness.goto();

  await page.setViewportSize({ width: 700, height: 700 });
});

describe('linkHref', () => {
  test.each([undefined, 'hash', 'omit'].map(linkHref => [linkHref]))(
    'Setting linkHref to %p',
    async previousLabel => {
      await testHarness.setField('linkHref', previousLabel);

      await testHarness.waitForNextFrame();

      const paginationHtml = await testHarness.getPaginationHtml();

      expect(paginationHtml).toMatchSnapshot();
    },
  );

  test('Setting linkHref to a function', async () => {
    await testHarness.setField('linkHref', 'hrefTestFn()');

    await testHarness.waitForNextFrame();

    const paginationHtml = await testHarness.getPaginationHtml();

    expect(paginationHtml).toMatchSnapshot();
  });
});
