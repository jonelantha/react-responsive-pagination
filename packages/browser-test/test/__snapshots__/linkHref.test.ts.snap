// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`linkHref Setting linkHref to "hash" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#">1<span class="sr-only"> (current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`linkHref Setting linkHref to "omit" 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link">1<span class="sr-only"> (current)</span></a></li><li class="page-item"><a class="page-link">2</a></li><li class="page-item"><a class="page-link">3</a></li><li class="page-item"><a class="page-link">4</a></li><li class="page-item"><a class="page-link">5</a></li><li class="page-item"><a class="page-link">6</a></li><li class="page-item"><a class="page-link">7</a></li><li class="page-item"><a class="page-link">8</a></li><li class="page-item"><a class="page-link">9</a></li><li class="page-item"><a class="page-link">10</a></li><li class="page-item"><a class="page-link">11</a></li><li class="page-item"><a class="page-link">12</a></li><li class="page-item"><a class="page-link">13</a></li><li class="page-item"><a class="page-link">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link">100</a></li><li class="page-item"><a class="page-link" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;

exports[`linkHref Setting linkHref to undefined 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span><span class="sr-only">Previous</span></span></li><li class="page-item active"><a class="page-link" href="#">1<span class="sr-only"> (current)</span></a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">3</a></li><li class="page-item"><a class="page-link" href="#">4</a></li><li class="page-item"><a class="page-link" href="#">5</a></li><li class="page-item"><a class="page-link" href="#">6</a></li><li class="page-item"><a class="page-link" href="#">7</a></li><li class="page-item"><a class="page-link" href="#">8</a></li><li class="page-item"><a class="page-link" href="#">9</a></li><li class="page-item"><a class="page-link" href="#">10</a></li><li class="page-item"><a class="page-link" href="#">11</a></li><li class="page-item"><a class="page-link" href="#">12</a></li><li class="page-item"><a class="page-link" href="#">13</a></li><li class="page-item"><a class="page-link" href="#">14</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span><span class="sr-only">Next</span></a></li></ul>"`;
